onnxscript/rewriter/fuse_pad_into_conv_test.py

```diff
@@ -0,0 +1,332 @@
+
+import numpy as np
+import onnx_ir as ir
+import parameterized
+from onnx_ir.passes.common import onnx_checker, shape_inference
+
+from onnxscript.rewriter import pattern as orp
+from onnxscript.rewriter import testing
+from onnxscript.rewriter.fuse_pad_into_conv import (
+    fuse_pad_into_conv,
+    fuse_pad_into_conv_rule_set,
+    normalize_pad_format_conv,
+)
+
+
+def _clone_model(model: ir.Model) -> ir.Model:
+    return ir.from_proto(ir.to_proto(model))
+
+
+class FusePadConvBaseTest(unittest.TestCase):
+    @property
+    def rng(self):
+        return np.random.default_rng(20250522)
+
+    def get_conv_weights(self, shape: typing.Sequence[int], tape: ir.tape.Tape = None):
+        w = ir.tensor(self.rng.uniform(-0.5, 0.5, shape).astype("float32"), name="W")
+        if tape is not None:
+            w = tape.initializer(w)
+        return w
+
+    def build_model(
+        self,
+        op_type: str,
+        input_shape: ir.Shape,
+        weight_shape: typing.Sequence[int],
+        pad_inputs: typing.Sequence[ir.TensorProtocol | ir.Value | None],
+        pad_attributes: typing.Mapping[str, ir.Attr] | None = None,
+        conv_attributes: typing.Mapping[str, ir.Attr] | None = None,
+        opset_imports: typing.Mapping[str, int] = {"": 20},
+    ) -> ir.Model:
+        tape = ir.tape.Tape()
+        inputs = []
+        output_shape = ir.Shape((input_shape[0],) + ("?",) * (len(input_shape) - 1))
+
+        # Convert pad_inputs to initializers (if needed)
+        pad_inputs = list(pad_inputs)
+        for idx, x in enumerate(pad_inputs):
+            if isinstance(x, ir.TensorProtocol):
+                pad_inputs[idx] = tape.initializer(x)
+            elif isinstance(x, ir.Value):
+                inputs.append(x)
+            elif isinstance(x, float):
+                pad_inputs[idx] = tape.op("Constant", inputs=[], attributes={"value_float": x})
+            elif x is not None:
+                raise ValueError(f"Unsupported type for pad input ({x}): {type(x)}.")
+
+        # Register operations in the tape
+        idtype = ir.DataType.UINT8 if op_type == "ConvInteger" else ir.DataType.FLOAT
+        x = ir.Input("X", shape=input_shape, type=ir.TensorType(idtype))
+        y = tape.op("Pad", inputs=[x, *pad_inputs], attributes=pad_attributes)
+        y = tape.op(
+            op_type,
+            inputs=[y, self.get_conv_weights(weight_shape, tape)],
+            attributes=conv_attributes,
+            output=ir.Input("Y", shape=output_shape, type=ir.TensorType(x.dtype)),
+        )
+        if op_type == "ConvInteger":
+            y.dtype = ir.DataType.INT32
+
+        # Build the model
+        ir_model = ir.Model(
+            ir.Graph(
+                inputs=[x, *inputs],
+                outputs=[y],
+                nodes=tape.nodes,
+                initializers=tape.initializers,
+                opset_imports=opset_imports,
+                name="model",
+            ),
+            ir_version=9,
```

Comment:
```suggestion
            ir_version=10,
```

nit: prefer 10 but doesn't matter too much